discord_slash/client.py

```diff
@@ -417,12 +415,19 @@ def add_subcommand(self,
         name = name or cmd.__name__
         name = name.lower()
         description = description or getdoc(cmd)
-        guild_ids = guild_ids if guild_ids else []
-
-        if base in self.commands:
-            for x in guild_ids:
-                if x not in self.commands[base].allowed_guild_ids:
-                    self.commands[base].allowed_guild_ids.append(x)
+        guild_ids = guild_ids if guild_ids else ["global"]
+
+        for i in range(len(guild_ids)):
+            if guild_ids[i] is None:
+                guild_ids[i] = "global"
+            if guild_ids[i] not in self.commands:
+                self.commands[guild_ids[i]] = {}
+            if guild_ids[i] not in self.subcommands:
+                self.subcommands[guild_ids[i]] = {}
+            if base in self.commands[guild_ids[i]]:
+                for x in guild_ids:
+                    if x not in self.commands[guild_ids[i]][base].allowed_guild_ids:
+                        self.commands[guild_ids[i]][base].allowed_guild_ids.append(x)
```

Comment:
Pretty sure this could cause issues
Because you could have for example:
```
/base
 - cmd1 - guild1
 - cmd2 - guild2
 - cmd3 - guild1 + guild2
 ```
 Not actually sure that it would but it seems like it might? (this is confusing)